From ac4105abd4cf2a0a791d26c8ea7c66cf961b6bb8 Mon Sep 17 00:00:00 2001
From: Xi Ruoyao <xry111@xry111.site>
Date: Tue, 3 Feb 2026 10:44:24 +0800
Subject: [PATCH 1/4] UPSTREAM: drop -t from mktemp

This option is deprecated and it makes an unwanted effect: if TMPDIR is
set in the environment it got used instead of the directory we are
specifying with -p.

I cannot see any benefit of it, so just drop it.

Closes: https://lists.linuxfromscratch.org/sympa/arc/blfs-dev/2026-02/msg00002.html
(cherry picked from commit 7d5aca91d6207a1d40101f439231f2f7341520df)
---
 make-ca | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/make-ca b/make-ca
index 437b1b9..d8c6da6 100644
--- a/make-ca
+++ b/make-ca
@@ -748,7 +748,7 @@ CERTBEGINLIST=`grep -n "^# Certificate" "${WORKDIR}/certdata.txt" | \
 
 # Dump individual certs to temp file
 for certbegin in ${CERTBEGINLIST}; do
-  tmpcrtbeg=$(mktemp -p ${TEMPDIR}/certs -t make-ca.XXXXXXXX.tmp)
+  tmpcrtbeg=$(mktemp -p ${TEMPDIR}/certs make-ca.XXXXXXXX.tmp)
   awk "NR==$certbegin,/^CKA_TRUST_STEP_UP_APPROVED/" "${WORKDIR}/certdata.txt" \
       > $tmpcrtbeg
 done
@@ -916,7 +916,7 @@ if test -d "${LOCALDIR}"; then
     write_anchor
 
     # Generate working copy
-    tempcert=$(mktemp -p ${TEMPDIR}/certs -t make-ca.XXXXXXXX.pem)
+    tempcert=$(mktemp -p ${TEMPDIR}/certs make-ca.XXXXXXXX.pem)
     "${OPENSSL}" x509 -in "${cert}" -text -fingerprint > ${tempcert}
 
     # Add to Shared NSS DB
-- 
2.56.0

